src/AI/InfantrySpamAI.java

```diff
@@ -222,9 +223,11 @@ public GameAction getNextAction(GameMap gameMap)
           if( loc.getEnvironment().terrainType == TerrainType.FACTORY && buyer.army == myArmy && loc.getResident() == null )
           {
             ArrayList<UnitModel> units = buyer.getShoppingList(loc);
-            if( !units.isEmpty() && buyer.getBuyCost(units.get(0), loc.getCoordinates()) <= myArmy.money )
+            final int buyCost = buyer.getBuyCost(units.get(0), loc.getCoordinates());
+            if( !units.isEmpty() && buyCost <= budget )
             {
               GameAction action = new GameAction.UnitProductionAction(buyer, units.get(0), loc.getCoordinates());
+              budget -= buyCost;
```

Comment:
Why is this needed? Is it just that ISAI could potentially go over budget if merged with another CO who used most/all of the budget already, and this would result in an invalid/turn-ending action?

Reply:
I was using "zombie apocalypse mode" (Cinder+ISAI) to test things, and noticed that ISAI would never use `Sear` because of invalid buy actions.